### ISSUE TITLE
[TW-548] Modifiers expanding even when in description

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#583]
_Michelle Crane on 2011-10-25T12:46:09Z says:_ 

 I understand how commands/modifiers in task can be reduced to their shortest unambiguous strings (e.g., pro: can be used instead of project:).  I've noticed that you can do the same thing with both 'recur' and 'wait'.  However, I'm finding that these last two are being expanded when they don't need to be.  

For instance, I often use 're' in my task description, e.g., 'talk to so-and-so re whatever'.  But when my task is added, the 're' is expanded to 'recur'.  The first time this happened, I thought I had just entered the task incorrectly.

I think that the expansion is great, but only if I'm using those commands as a command/modifier - I expect that 're:' would turn into 'recur:', but I don't expect that 're' should be 'recur'.  Or 'wa' the same as 'wait'.  In other words, the parser should be checking for a colon, not a space.

Just as a quick check of other commands, 'pro' is also expending to 'project' in the description, 'dep' to 'depends', 'stat' to 'status'.  I'm guessing that most/all of the commands will be expanded even when in the text description.

Note: this doesn't happen with task 1.9.4.

Trace of task 2.0.0 beta4, showing me adding a couple of tasks and getting unexpected results.

<pre>
F:\Dropbox\TextFiles>task add Talk to so-and-so re that thing
Created task 1.

F:\Dropbox\TextFiles>task add Travel to seattle, wa this year
Created task 2.

F:\Dropbox\TextFiles>task new

ID Project ! Due Count * + R Dep Description                        Age
* - ------- - --- ----- - - - --- ---------------------------------- ---
 2                               Travel to seattle, wait this year   2s
 1                               Talk to so-and-so recur that thing 15s
2 tasks

F:\Dropbox\TextFiles>task add This is a recurring task due:eow recur:weekly
Created task 3.

F:\Dropbox\TextFiles>task add This is another recurring task due:eom re:monthly
Created task 5.

F:\Dropbox\TextFiles>task new

ID Project ! Due        Count   * + R Dep Description                        Age
* - ------- - ---------- ------- - - - --- ---------------------------------- ---
 6           2011-10-31 -5 days     R     This is another recurring task       -
 4           2011-10-28 -2 days     R     This is a recurring task           27s
 2                                        Travel to seattle, wait this year  56s
 1                                        Talk to so-and-so recur that thing  1m
4 tasks

</pre>

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2011-10-25T12:46:09Z
Modified: 2014-02-09T02:04:41Z
```

**Comment by taskwarrior** [comment]
_Cory Donnelly on 2011-10-27T14:26:41Z says:_ 

 Great catch, thanks Michelle.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2012-03-03T21:13:21Z says:_ 

 Deferring this to some magical time in the future when the parser can shut this off.  In the meantime, this workaround exists:

<pre>
task add -- Talk to so-and-so re that thing
</pre>

**Comment by taskwarrior** [comment]
_Louis-Claude Canon on 2012-07-09T12:01:19Z says:_ 

 This was fixed when solving Bug #1006.
